src/tardev-snapshotter/src/snapshotter.rs

```diff
@@ -1012,16 +1012,79 @@ impl TarDevSnapshotter {
                         return Err(anyhow!(copy_error));
                     }
                 }
+                file.flush()?;
 
-                trace!("Appending index to {:?}", &base_name);
-                file.rewind().context("failed to rewind the file handle")?;
-                tarindex::append_index(&mut file).context("failed to append tar index")?;
+                // Create an erofs metadata using mkfs.erofs
+                let layer_path = base_name.with_extension("erofs");
+                debug!(
+                    "Creating erofs meta image {:?} from {:?}",
+                    &layer_path, &base_name
+                );
+                let status = Command::new("mkfs.erofs")
+                    .args([
+                        "--tar=i",
+                        "-T", "0", // zero out unix time
+                        "-U", "c1b9d5a2-f162-11cf-9ece-0020afc76f16", // set UUID to something specific
+                        "--quiet",
+                        layer_path.to_str().unwrap(),
+                        base_name.to_str().unwrap(),
+                    ])
+                    .status()
+                    .context("Failed to execute mkfs.erofs command")?;
+
+                if !status.success() {
+                    return Err(anyhow!(
+                        "mkfs.erofs failed with status: {}",
+                        status.code().unwrap_or(-1)
+                    ));
+                }
 
-                trace!("Appending dm-verity tree to {:?}", &base_name);
-                let root_hash = verity::append_tree::<Sha256>(&mut file)
+                // Append the decompressed tar file to the erofs metadata
+                debug!(
+                    "Appending decompressed tar file {:?} to erofs metadata {:?}",
+                    &base_name, &layer_path
+                );
+                let mut erofs_file = OpenOptions::new()
+                    .write(true)
+                    .append(true)
+                    .open(&layer_path)
+                    .context("failed to open erofs metadata for appending")?;
+                let mut base_file = File::open(&base_name)
+                    .context("failed to open decompressed tar file for reading")?;
+                std::io::copy(&mut base_file, &mut erofs_file)
+                    .context("failed to append decompressed tar file to erofs metadata")?;
+                
+                // get size of erofs meta + tar
+                let erofs_file_size = erofs_file.metadata()
+                    .expect("Failed to get metadata")
+                    .len();
+
+                // Align the size to 512 bytes
+                let alignment = 512;
```

Comment:
Would be good to have this as a common constant somewhere. Fine for now.

Reply:
Created an erofs-common package that has the alignment and uuid as constants.